File: spring_java_orm_sample/src/main/java/com/example/demo/jooq/entity/BookWithStockRow.java
package com.example.demo.jooq.entity;

import lombok.Value;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Value
public class BookWithStockRow {
    Long id;
    String title;
    String author;
    LocalDate releaseDate;
    Long publisherId;
    String publisherName;
    Long genreId;
    String genreName;
    String isbn;
    LocalDateTime updateAt;
    Long version;
    Long bookStockId;
    Long bookStockStoreId;
    String storeName;
    Integer bookStockQuantity;
}

File: codex_poting/kotlin/src/main/kotlin/com/example/demo/jooq/entity/BookWithStockRow.kt
package com.example.demo.jooq.entity

import java.time.LocalDate
import java.time.LocalDateTime

data class BookWithStockRow(
    val id: Long?,
    val title: String?,
    val author: String?,
    val releaseDate: LocalDate?,
    val publisherId: Long?,
    val publisherName: String?,
    val genreId: Long?,
    val genreName: String?,
    val isbn: String?,
    val salesUnitPrice: Int?,
    val updateAt: LocalDateTime?,
    val version: Long?,
    val bookStockId: Long?,
    val bookStockStoreId: Long?,
    val storeName: String?,
    val bookStockQuantity: Int?
)
